src/BusinessLayer/TelemetryReporting.h

```diff
@@ -33,29 +35,58 @@ class TelemetryReporting : public QObject
                        const LightsData& lightsData_,
                        View& view);
 
+    friend class TelemetryReportingTest;
```

Comment:
I don't really get how this is suppose to help us here. To test the correct behavior it is required to call the private function in order to check if the mocked communication service is called correctly. I don't see how we could avoid this by using mocked data classes.
However creation of mocked data classes make sense as soon as they provide additional methods. So far they just store public data. Therefore one could simply initialize the used fields explicitely in the setUp method. Do you think I should do that?